src/e3/anod/qualifiers_manager.py

```diff
@@ -0,0 +1,799 @@
+            make the help/error clearer.
+        :param test_only: By default the qualifiers are used by all anod actions
+            (install, build, test...). If test_only is True, then this qualifier is
+            only available for test.
+        :param repr_alias: An alias for the qualifier name used by the name generation.
+            By default, the repr_alias is the qualifier name itself.
+        :param repr_in_hash: False by default. If True, the qualifier is included in
+            the hash at the end of the generated name. The result is less readable but
+            shorter.
+        :return: A QualifierDecalration instance.
+        """
+        if self.is_declaration_phase_finished:
+            raise AnodError(
+                "The qualifier declaration is finished. It is not possible to declare a"
+                " new qualifier after a component has been declared or a name generated"
+            )
+
+        new_qualifier: QualifierDeclaration = {
+            "description": description.strip(),
+            "test_only": test_only,
+            "repr_alias": name if repr_alias is None else repr_alias.strip(),
+            "repr_in_hash": repr_in_hash,
+        }
+        # Check all the inputs.
+
+        # Check name:
+        name = name.strip()
+        if not name:
+            raise AnodError("The qualifier name cannot be empty")
+        elif name in self.qualifier_decls:
+            raise AnodError(f"The {name} qualifier has already been declared")
+
+        # Check description:
+        if not new_qualifier["description"]:
+            raise AnodError(f"The {name} qualifier description cannot be empty")
+
+        # Check repr_alias:
+        if new_qualifier["repr_alias"] is not None and not new_qualifier["repr_alias"]:
+            raise AnodError(f"The {name} qualifier repr_alias cannot be empty")
+
+        return new_qualifier
+
+    def declare_tag_qualifier(
+        self,
+        name: str,
+        description: str,
+        test_only: bool = False,
+        repr_alias: str | None = None,
+        repr_in_hash: bool = False,
+    ) -> None:
+        """Declare a new tag qualifier.
+
+        From the user prospective, 'declare' mean that a qualifier can be used in the
+        spec and will have an impact on the build space name.
+
+        From an internal point of view, this mean that the qualifier is added to
+        the qualifier_decls instance variable.
+
+        The tag qualifier are qualifier with an implicit value. Their value is True if
+        the qualifier is passed at runtime and False else.
+
+        This method cannot be called after the end of the declaration phase.
+
+        :param name: The name of the qualifier. It used to identify it and pass it to
+            the spec.
+        :param description: A description of the qualifier purposes. It is used to
+            make the help/error clearer.
+        :param test_only: By default the qualifier are used by all anod actions
```

Comment:
Just a remark, here, maybe we can use an enum, BUILD_ONLY, TEST_ONLY, ALL or something like that to be less specific

Reply:
But it is just a suggestion, no need to change that now could be done in a second time